### PR TITLE
Development installer

### PR DESCRIPTION
**Pull request by despiegk**
Resolves: https://github.com/threefoldtech/jumpscaleX/issues/515

### Before asking for approval

- [ ] Ensure the branch is up to date with its base branch.
- [ ] Corresponding changes to the documentation were made.

### CODE REVIEW

File: Jumpscale/data/nacl/NACL.py [hunk 1]
```diff
@@ -112,46 +117,26 @@ def configure(self, privkey_words=None, secret=None, sshagent_use=None, interact
...
 
-        j.application.interactive = j.application.interactive or interactive
-
-        # create dir where the secret will be to encrypt the secret
-        j.sal.fs.createDir(j.core.tools.text_replace("{DIR_VAR}/logs"))
-
-        j.sal.fs.remove(self._path_encryptor_for_secret)
-
-        redis_key = "secret_%s" % self.name
+        j.shell()
```
Comment (rkhamis) [inline Jumpscale/data/nacl/NACL.py]: Debuggers?

File: Jumpscale/tools/console/Console.py [hunk 2]
```diff
@@ -275,6 +274,10 @@ def askString(self, question, defaultparam="", regex=None, retry=-1, validate=No
             % (retry, regex)
         )
 
+    def _check_interactive(self):
+        if not j.core.myenv.interactive:
+            raise j.exceptions.Input("Cannot use console in a non interactive mode.", "console.noninteractive")
```
Comment (rkhamis) [inline Jumpscale/tools/console/Console.py:279]: shouldn't checks be True/False?

**Review by rkhamis — CHANGES_REQUESTED** [review]
I've been trying to use this branch to install.
I tried using jsx as my entry point with `python3 jsx`, which first gives a stacktrace until configure is called(?)

after trying configure (which also didn't work) and manually generating a new key and loading it, I re-configured then tried to finally install. It fails now on 

```

EXCEPTION

        jsx.py                                                       : 414 : cli()
        /usr/local/lib/python3.6/dist-packages/click/core.py         : 764 : return self.main(*args, **kwargs)
        /usr/local/lib/python3.6/dist-packages/click/core.py         : 717 : rv = self.invoke(ctx)
        /usr/local/lib/python3.6/dist-packages/click/core.py         : 1137: return _process_result(sub_ctx.command.invoke(sub_ctx))
        /usr/local/lib/python3.6/dist-packages/click/core.py         : 956 : return ctx.invoke(self.callback, **ctx.params)
        /usr/local/lib/python3.6/dist-packages/click/core.py         : 555 : return callback(*args, **kwargs)
        jsx.py                                                       : 243 : installer.install(config={}, sandboxed=False, force=force, gitpull=pull)
        code/github/threefoldtech/jumpscaleX/install/InstallTools.py : 2533: BaseInstaller.install(basedir=basedir, config=config, sandboxed=sandboxed, force=force)
        code/github/threefoldtech/jumpscaleX/install/InstallTools.py : 2114: MyEnv.init(basedir=basedir, config=config, readonly=False, force=True)

Wed 22 12:21:45 core.py          - 762 -                                    : EXCEPTION: init() got an unexpected keyword argument 'basedir'
CANNOT CONTINUE

```

**Comment by zgorizzo69** [comment]
`python3 jsx.py container-install`
followed by 
`python3 jsx.py container-kosmos`
worked for me 
it creates a 3bot container and then launch kosmos

**Comment by zgorizzo69** [comment]
I have tried with the latest commit `python3 install/jsx.py container-install -d` now I am asked twice for 
provide secret to use for encrypting private key: 
If you don't enter anything it accepts it and at the end fails so I have committed a change for that
but we shouldn't be asked twice for the secret to encrypt the private key ( Note that even if the second secret is different from the first one it goes on)

**Pull request merged**